tests/e2e/specs/activate-and-setup/setup-wizard.test.js

```diff
@@ -1,35 +1,240 @@
+import toClick from 'expect-puppeteer/lib/matchers/toClick';
+import {
+	adminUrl,
+	cleanupSenseiData,
+	resetSetupWizard,
+} from '../../utils/helpers';
 import { AdminFlow } from '../../utils/flows';
 
-describe( 'Admin can login and make sure Sensei LMS is activated', () => {
-	it( 'Should login', async () => {
-		await AdminFlow.login();
-		const adminMenuMain = await page.waitForSelector( '#adminmenumain' );
+async function openSetupWizard() {
+	return AdminFlow.goTo( 'admin.php?page=sensei_setup_wizard' );
+}
+
+async function stepIsComplete( label ) {
+	return expect( page ).toMatchElement(
+		'.woocommerce-stepper__step.is-complete',
+		{
+			text: label,
+			timeout: 5000,
+		}
+	);
+}
+
+async function stepIsActive( label ) {
+	return expect( page ).toMatchElement(
+		'.woocommerce-stepper__step.is-active',
+		{
+			text: label,
+		}
+	);
+}
 
-		expect( adminMenuMain ).toBeTruthy();
+/**
+ * Setup Wizard E2E tests.
+ *
+ * These tests should be run sequentially.
+ */
+describe( 'Setup Wizard', () => {
+	beforeAll( async () => {
+		await AdminFlow.login();
 	} );
 
-	it( 'Should make sure Sensei LMS is activated or activate it', async () => {
-		const slug = 'sensei-lms';
-		let deactivateLink;
+	it( 'opens when first activating the Sensei LMS plugin', async () => {
+		await AdminFlow.activatePlugin( 'sensei-lms' );
+		await cleanupSenseiData();
+
+		await AdminFlow.activatePlugin( 'sensei-lms', true );
+		await expect( page.url() ).toMatch(
+			'admin.php?page=sensei_setup_wizard'
+		);
+	} );
 
+	it( 'shows a notice to run the Setup Wizard', async () => {
 		await AdminFlow.goToPlugins();
 
-		deactivateLink = await page.$(
-			`tr[data-slug="${ slug }"] .deactivate a`
-		);
-		if ( deactivateLink ) {
-			return;
-		}
+		await expect( page ).toClick( '.sensei-message a', {
+			text: 'Setup Wizard',
+		} );
+		await page.waitForNavigation();
+		await expect( page ).toMatch( 'Welcome to Sensei LMS!' );
+	} );
 
-		await page.click( `tr[data-slug="${ slug }"] .activate a` );
+	describe( 'Welcome step', () => {
+		beforeAll( async () => await resetSetupWizard() );
 
-		// Go to the plugins page again to make sure it's in the plugin page.
-		// It can be the setup wizard.
-		await AdminFlow.goToPlugins();
-		deactivateLink = await page.waitForSelector(
-			`tr[data-slug="${ slug }"] .deactivate a`
-		);
+		it( 'opens on first launch', async () => {
+			await openSetupWizard();
+			await expect( page ).toMatch( 'Welcome to Sensei LMS!' );
+		} );
+
+		it( 'displays usage tracking modal when clicking continue', async () => {
+			await expect( page ).toClick( 'button', { text: 'Continue' } );
+			await expect( page ).toMatch( 'Build a Better Sensei LMS' );
+		} );
+
+		it( 'allows opting in', async () => {
```

Comment:
WDYT in the end of this test navigate to the settings and check if the usage tracking checkbox is checked?

Reply:
I'll bring add `Sensei_Test_Events` in a new PR after this and the mocking one are merged to prevent the tracking requests, and add a query here to check if the event was triggered.